### DISCUSSION TITLE
Planet rendering library

### DISCUSSION THREAD

**Discussion by mattiaverga** [CelestiaProject/Celestia#1964] [Ideas]
Maybe I'm going really out of scope of Celestia, but I'll try to drop this idea.
I was just looking at how many planetarium software rely on the old [XPlanet](https://sourceforge.net/projects/xplanet/) software for rendering planets in their UI.
Since XPlanet development doesn't seem to be much active, could Celestia be a valid alternative, maybe exposing a library for external programs to render planets surfaces? How scientific accurately does Celestia renders planet surfaces (meaning Earth-planet relative viewpoint, Jupiter GMR longitude settings, etc.)?

**Comment by 375gnu** [comment]
Celestia takes into account only non-spherical form of bodies, in 1.7 we fixed (not completely, there're still ways to break) object positions on surface.
